### PR TITLE
NET-5398: V2 unavailable UI message

### PR DESCRIPTION
**Pull request by WenInCode**
### Description

**This is currently based off #20353 until that gets merged**

Adds a redirect to a `/unavailable` page that shows a message about the UI being unvailable if you have v2 experiment enabled. It also redirects away from the unavailable page if it is not enabled.

Screenshot 2024-01-26 at 9 42 18 AM
Screenshot 2024-01-26 at 9 40 44 AM

### Testing & Reproduction steps

I mainly tested this by building Consul and enabling the experiment in the server config and then opening the UI ([steps in this ticket](https://hashicorp.atlassian.net/browse/CC-6956)).

You can do the naïve test as well by running this against the mocks and then running `Scenario('CONSUL_V2_CATALOG_ENABLE=1')` in the console.

### Links

#20353 
https://hashicorp.atlassian.net/browse/NET-5398
https://hashicorp.atlassian.net/browse/CC-6956

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: .changelog/20353.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
+ui: adds V2CatalogEnabled to config that is passed to the ui
```
Comment (chris-hut) [inline .changelog/20353.txt]: https://github.com/hashicorp/consul/blob/main/.changelog/20353.txt
This file does exist, I think you maybe need to pull in main?

File: ui/packages/consul-ui/app/routes/application.js [hunk 2]
```diff
@@ -13,13 +13,21 @@ export default class ApplicationRoute extends Route.extend(WithBlockingActions)
...
 
+  beforeModel() {
+    if (this.env.var('CONSUL_V2_CATALOG_ENABLED')) {
+      this.router.replaceWith('unavailable');
+    }
+  }
+
   async model() {
-    if (this.env.var('CONSUL_ACLS_ENABLED')) {
+    if (this.env.var('CONSUL_ACLS_ENABLED') && !this.env.var('CONSUL_V2_CATALOG_ENABLED')) {
```
Comment (chris-hut) [inline ui/packages/consul-ui/app/routes/application.js:27]: Do we need this check - `CONSUL_V2_CATALOG_ENABLED` always be false here?
Won't the `beforeModel` take us to `unavailable`?

File: ui/packages/consul-ui/app/templates/unavailable.hbs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+              @isHrefExternal={{true}}
+              @text='Provide feedback'
+              @href='https://hashicorp.sjc1.qualtrics.com/jfe/form/SV_cHDaObigIBbQneC'
+            />
+          </A.Footer>
+        </Hds::ApplicationState>
+      </div>
+    </div>
+  {{else}}
+    {{did-insert (route-action 'replaceWith' 'index')}}
```
Comment (chris-hut) [inline ui/packages/consul-ui/app/templates/unavailable.hbs:106]: Don't we also do this in the `route`?

**Review by chris-hut — APPROVED** [review]
Nice!
Just a couple questions about doubling down

File: ui/packages/consul-ui/app/templates/unavailable.hbs (hunk 3 above)
Comment (WenInCode) [reply ui/packages/consul-ui/app/templates/unavailable.hbs:106]: You are very correct, I forgot to remove this once I added the route. I will do that on Monday. Thanks!

File: ui/packages/consul-ui/app/routes/application.js (hunk 2 above)
Comment (WenInCode) [reply ui/packages/consul-ui/app/routes/application.js:27]: Ya you are correct. Good suggestion, I'll update it on Monday :)

**Pull request merged**